src/metpy/plots/declarative.py

```diff
@@ -656,6 +660,39 @@ class MapPanel(Panel):
     'xx-small', 'x-small', 'small', 'medium', 'large', 'x-large', 'xx-large'.
     """
 
+    @validate('area')
+    def _valid_area(self, proposal):
+        """Check that proposed string or tuple is valid and turn string into a tuple extent."""
+        area = proposal['value']
+
+        # Parse string, check that string is valid, and determine extent based on string
+        if isinstance(area, str):
+            try:
+                region, modifier = re.match(r'(\w+)([-+]*)$', area).groups()
+            except AttributeError:
+                raise TraitError('"' + area + '" is not a valid string area.')
+
+            if region.lower() not in _areas.keys() and region != 'global':
+                raise TraitError('"' + area + '" is not a valid string area.')
+            elif region == 'global':
+                extent = 'global'
+            else:
+                extent = _areas[region.lower()]
+                zoom = modifier.count('+') - modifier.count('-')
+                extent = self._zoom_extent(extent, zoom)
```

Comment:
I think if we re-order this can become a bit simpler (plus we can drop `keys()` and use a f-string):
```suggestion
            region = region.lower()
            if region == 'global':
                extent = 'global'
            elif region in _areas:
                extent = _areas[region]
                zoom = modifier.count('+') - modifier.count('-')
                extent = self._zoom_extent(extent, zoom)
            else:
                raise TraitError(f'"{area}" is not a valid string area.')
```
This also makes us accept "GLOBAL" as a case-insensitive area, which matches the other areas.